### DISCUSSION TITLE
Can't embed SVG from Releases into README

### DISCUSSION THREAD

**Discussion by itsmattkc** [community/community#59781] [Repositories]
### Select Topic Area

Bug

### Body

My CI generates an SVG that details the overall status of the project that I want to embed into the README.md. To make the SVG available, the CI uploads it as a "continuous release", which gives it a consistent URL. However, when trying to embed that SVG into the readme, it fails, showing an "error icon". Notably this does **not** happen with PNGs or other types of raster images uploaded as a release asset, only SVGs, but raster images produce significantly worse results for what is a lot of text.

The issue _seems_ to be that GitHub release assets are provided with the `application/octet-stream` content type instead of `image/svg+xml`. I'm guessing this is to persuade the browser to download files rather than open them itself, but it seems to prevent the browser from detecting the SVG as an image at all.

Is there any way to get around this or perhaps alternative way or making these SVGs available? I'm trying to avoid having to rely on an external host to upload to/embed from. I can't upload them as an "actions artifact" because they get auto-zipped.

**Comment by ArdeshirV** [comment]
If you use this kind of statement:
```![sample SVG image](https://github.com/ArdeshirV/sample/blob/master/gplv3.svg)```

You can also use this one to see the difference, and it usually works:
```<img src="https://github.com/ArdeshirV/sample/blob/master/gplv3.svg" >```

It's better you give us a reference to your README.md file and your SVG-file, then we can see the problem more clear and find the solution.

**Comment by itsmattkc** [comment]
I'm aware of both of those syntaxes, and they work in cases like the one you've written where the SVG is in the _repository_, but not when the SVG is part of a **release**. The SVG I want to embed is auto-generated on the CI and uploaded as a **release**, it is not a static file in the repository.

I can link you an example URL I want to embed (https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG), but my README isn't currently using that link since it doesn't embed properly. Right now, I'm uploading it to an external host as well as releases; a suboptimal solution, but that embeds just fine, so I know it's not the SVG itself. Like I said, it appears to be because the server provides `application/octet-stream` as the content type.

**Comment by ArdeshirV** [comment]
Now I can understand you better, I still don't know what is your external host web-server, but I know something like this can correct the problem on your host:

Add this to your .htaccess file at the root to fix it:
``` AddType image/svg+xml .svg .svgz  ```


for another example, if you use Apache web server, you can change APACHECONF like this:
```
<IfModule mod_expires.c>
  # Data interchange
    AddType application/atom+xml      atom
    AddType application/json          json map topojson
    AddType application/ld+json       jsonld
    AddType application/rss+xml       rss
    AddType application/geo+json      geojson
    AddType application/rdf+xml       rdf
    AddType application/xml           xml
  # JavaScript
    AddType text/javascript           js mjs
  # Manifest files
    AddType application/manifest+json     webmanifest
    AddType application/x-web-app-manifest+json         webapp
    AddType text/cache-manifest           appcache
  # Media files
    AddType audio/mp4                     f4a f4b m4a
    AddType audio/ogg                     oga ogg opus
    AddType image/bmp                     bmp
    AddType image/svg+xml                 svg svgz
    AddType image/webp                    webp
    AddType video/mp4                     f4v f4p m4v mp4
    AddType video/ogg                     ogv
    AddType video/webm                    webm
    AddType image/x-icon    cur ico
  # HEIF Images
    AddType image/heic                    heic
    AddType image/heif                    heif
  # HEIF Image Sequence
    AddType image/heics                   heics
    AddType image/heifs                   heifs
  # AVIF Images
    AddType image/avif                    avif
  # AVIF Image Sequence
    AddType image/avis                    avis
  # WebAssembly
    AddType application/wasm              wasm
  # Web fonts
    AddType font/woff                         woff
    AddType font/woff2                        woff2
    AddType application/vnd.ms-fontobject                eot
    AddType font/ttf                          ttf
    AddType font/collection                   ttc
    AddType font/otf                          otf
  # Other
    AddType application/octet-stream          safariextz
    AddType application/x-bb-appworld         bbaw
    AddType application/x-chrome-extension    crx
    AddType application/x-opera-extension     oex
    AddType application/x-xpinstall           xpi
    AddType text/calendar                     ics
    AddType text/markdown                     markdown md
    AddType text/vcard                        vcard vcf
    AddType text/vnd.rim.location.xloc        xloc
    AddType text/vtt                          vtt
    AddType text/x-component                  htc
</IfModule>
```

**Comment by itsmattkc** [comment]
Sorry, you're still misunderstanding me. My external host is fine, but I'm only using it because embedding from GitHub Releases doesn't work (when I say "the server" provides the wrong content type, I mean GitHub's, not mine). As I said in the OP, "I'm trying to avoid having to rely on an external host to upload to/embed from".

To help understand better, here are steps to reproduce:

1. On any GitHub repository, create a "release".
2. Upload an SVG as an "asset" to that release (so it's listed as a file).
3. Attempt to embed that SVG in README.md.

Alternatively, try embedding the URL I listed earlier: https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG

If I try to embed that URL anywhere on GitHub, you can see it fails to work:

![SVG image](https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG)

**Reply by akc3n** [reply]
SVG files served via GitHub release URLs are proxied through `camo.githubusercontent.com`, which blocks certain content due to CSP and CORB (Cross-Origin Read Blocking) restrictions.
Based on the linked svg in your comment, when you try to open the SVG directly in the browser on this page.

As a result, using either of the following:
`![SVG image](https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG)`
![SVG image](https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG)
or
`<img src="https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG" alt="SVG image">`
<img src="https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG" alt="SVG image">
will fail to display the image and you'll see this error in the console:

```
Response was blocked by CORB (Cross-Origin Read Blocking)
Cross-Origin Read Blocking (CORB) blocked a cross-origin response.
1 request
 ISLEPROGRESS.SVG
```

So, instead of referencing the SVG directly from GitHub release assets like you attempted to do, try using this (as GitHub’s image proxy and embeds the image with a fallback canonical source, ensuring it renders correctly under GitHub’s CSP constraints):

```
<img src="https://camo.githubusercontent.com/d27978e87b3d5547e472ae3f6610ae64892cffc1ab7628ed0b8e683142d2bf8e/68747470733a2f2f6c65676f69736c616e642e6f72672f70726f67726573732f49534c4550524f47524553532e535647" data-canonical-src="https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG" style="max-width: 50%;">
```

Resulting in:
<img src="https://camo.githubusercontent.com/d27978e87b3d5547e472ae3f6610ae64892cffc1ab7628ed0b8e683142d2bf8e/68747470733a2f2f6c65676f69736c616e642e6f72672f70726f67726573732f49534c4550524f47524553532e535647" data-canonical-src="https://github.com/isledecomp/isle/releases/download/continuous/ISLEPROGRESS.SVG" style="max-width: 50%;">

**Reply by itsmattkc** [reply]
That does look like it works, but how does one retrieve that `https://camo.githubusercontent.com/d27978e87b3d5547e472ae3f6610ae64892cffc1ab7628ed0b8e683142d2bf8e/68747470733a2f2f6c65676f69736c616e642e6f72672f70726f67726573732f49534c4550524f47524553532e535647` link? What does it mean? Is that just a fixed URL that works for any SVG?

**Comment by garyo** [comment]
I'm having the same problem. My CI produces an SVG file as a release asset, and I can't embed it in my README. Even with `?sanitize=true` and with either syntax, Markdown image or HTML `<img .../>`.
